### DISCUSSION TITLE
There is no problem in local, and there is no error in vercel build, but 404 error

### DISCUSSION THREAD

**Discussion by suhvin** [vercel/next.js#52892] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.1.0: Sun Oct  9 20:14:30 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T8103
    Binaries:
      Node: 19.3.0
      npm: 9.2.0
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.4.1
      eslint-config-next: 13.4.1
      react: 18.2.0
      react-dom: 18.2.0
```

### Link to the code that reproduces this issue or a replay of the bug

https://firebasestorage.googleapis.com/v0/b/yeonpick-next.appspot.com/o/%E1%84%92%E1%85%AA%E1%84%86%E1%85%A7%E1%86%AB%20%E1%84%80%E1%85%B5%E1%84%85%E1%85%A9%E1%86%A8%202023-07-19%20%E1%84%8B%E1%85%A9%E1%84%92%E1%85%AE%2011.57.17.mov?alt=media

### To Reproduce

There is no problem in local, and there is no error in vercel build, but 404 error. Cause not found

### Describe the Bug

I am working with next and typescript and hosting with vercel. In local, it is running well without any errors, and I made it automatically hosted by connecting the git dev branch to vercel, but the hosting works fine, but suddenly the hosting does not work. When building, the error did not appear and went well, but when entering the domain provided by the vercel, [404
This page could not be found.] floats. Looking at the build log, I confirmed that the url hosted in the dev branch state a few days ago shows up without any problems. In the meantime, there seems to be a problem in the code, but I'm struggling because there is no way to debug it. There is no problem in local and there is no build error, but I want to know the cause because 404 is popping up.

### Expected Behavior

Like local, or like a few days ago, I hope the proper web page pops up without 404 popping up.

### How are you deploying your application? (if relevant)

vercel

**Comment by factsfinder** [comment]
This normally happens if you don't select the framework in vercel. Go to settings => general => build and development settings, and select nextjs. And redeploy. That should fix it.

**Comment by javierdebug** [comment]
I confirm this is happening as well for us in a project. It seems like a caché issue since it works for a while and on some pages.
This has been an issue in the past:
https://github.com/vercel/vercel/issues/3294
https://github.com/vercel/next.js/issues/34006
